### PR TITLE
Expose Host Load for Java API through JNI

### PR DESCRIPTION
**Pull request by Jean-Emile**

### CODE REVIEW

File: examples/java/CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +33,7 @@ endif()
 
 foreach (example app-bittorrent app-centralizedmutex app-masterworker app-pingpong app-tokenring async-yield async-waitall async-dsend
          cloud-migration cloud-masterworker dht-chord dht-kademlia energy-consumption energy-pstate energy-vm io-file io-storage
-         process-kill process-migration process-startkilltime process-suspend task-priority trace-pingpong)
+         process-kill process-migration process-startkilltime process-suspend task-priority trace-pingpong hostload)
```
Comment (mquinson) [inline examples/java/CMakeLists.txt]: please keep this list in alphabetical order

File: examples/java/CMakeLists.txt [hunk 2]
```diff
@@ -72,7 +73,7 @@ set(xml_files     ${xml_files}     ${CMAKE_CURRENT_SOURCE_DIR}/app/bittorrent/bi
 if(enable_java)
   foreach (example app-bittorrent app-centralizedmutex app-masterworker app-pingpong app-tokenring async-yield async-waitall async-dsend
            cloud-migration cloud-masterworker dht-chord dht-kademlia energy-consumption energy-pstate energy-vm io-file io-storage
-           process-kill process-migration process-startkilltime process-suspend task-priority trace-pingpong)
```
Comment (mquinson) [inline examples/java/CMakeLists.txt]: This list too is in alphabetical order

File: src/bindings/java/org/simgrid/msg/Host.java [hunk 3]
```diff
@@ -143,7 +143,13 @@ public boolean hasData() {
 	 * the value will be updated in kernel mode before returning the control to the requesting actor.
 	 */
 	public native double getConsumedEnergy();
-	
+
+	public native double getCurrentLoad();
```
Comment (mquinson) [inline src/bindings/java/org/simgrid/msg/Host.java]: Please add a little javadoc chunk to these new calls

File: src/bindings/java/org/simgrid/msg/Msg.java [hunk 4]
```diff
@@ -41,6 +41,7 @@ private Msg() {
 	/** Tell the kernel that you want to use the energy plugin */
 	public static final native void energyInit();
 	public static final native void fileSystemInit();
+	public static final native void loadInit();
```
Comment (mquinson) [inline src/bindings/java/org/simgrid/msg/Msg.java]: Please add the missing javadoc. And while you're at it, you could add it to the function right before ;)

File: src/msg/msg_legacy.cpp [hunk 5]
```diff
@@ -297,6 +297,11 @@ sg_host_t MSG_host_self()
 {
   return sg_host_self();
 }
+
+double MSG_host_get_load(sg_host_t host)
```
Comment (mquinson) [inline src/msg/msg_legacy.cpp]: There were no need to extend MSG. The Java API could have used the S4U API directly. But your change is better than expected here, don't change it.

File: examples/java/CMakeLists.txt (hunk 1 above)
Comment (Jean-Emile) [reply examples/java/CMakeLists.txt]: done

File: examples/java/CMakeLists.txt (hunk 2 above)
Comment (Jean-Emile) [reply examples/java/CMakeLists.txt]: done

File: include/simgrid/host.h [hunk 6]
```diff
@@ -93,6 +93,11 @@ XBT_PUBLIC double sg_host_get_available_speed(sg_host_t host);
 
 XBT_PUBLIC int sg_host_core_count(sg_host_t host);
 
+/** \ingroup m_host_management
+ * \brief Returns the current computation load (in flops per second).
```
Comment (Jean-Emile) [inline include/simgrid/host.h]: TODO:  replace  \ingroup -> @ingroup

File: src/bindings/java/org/simgrid/msg/Host.java (hunk 3 above)
Comment (Jean-Emile) [reply src/bindings/java/org/simgrid/msg/Host.java]: done

File: src/bindings/java/org/simgrid/msg/Msg.java (hunk 4 above)
Comment (Jean-Emile) [reply src/bindings/java/org/simgrid/msg/Msg.java]: yes of course.

File: ChangeLog [hunk 7]
```diff
@@ -56,6 +56,7 @@ SMPI:
 Java:
  - Due to an internal bug, Msg.run() must now be your last line.
    We hope to fix it in a future release, and we are sorry for the inconvenience.
+ - Expose host load plugin (i.e. loadInit, getCurrentLoad, getComputedFlops,getAvgLoad)
```
Comment (mquinson) [inline ChangeLog]: Could you please put this stanza in the changelog of the upcoming 3.22, not under the already released 3.21 ? ;)
Yes, that'd be the first documented change in 3.22, but we released last week or so.

Comment (mquinson) [inline ChangeLog]: Could you please put this stanza in the changelog of the upcoming 3.22, not under the already released 3.21 ? ;)
Yes, that'd be the first documented change in 3.22, but we released last week or so.

**Comment by mquinson** [comment]
Well, forget it. I merged it and will polish the last few things myself. Many thanks for your contribution, welcome onboard ;)

**Pull request merged**